lib/google_maps_geocoder/google_maps_geocoder.rb

```diff
@@ -96,11 +97,28 @@ def http(uri)
 
   def json_from_url(url)
     uri = URI.parse query_url(url)
+
     logger.debug('GoogleMapsGeocoder') { uri }
+
     response = http(uri).request(Net::HTTP::Get.new(uri.request_uri))
     ActiveSupport::JSON.decode response.body
   end
 
+  def handle_exception(google_response_json)
+    status = google_response_json['status']
+    # no exception
+    return if status == 'OK'
+
+    message = GoogleMapsGeocoderErrorHandler::GeneralGoecodingError\
+              .new(google_response_json).message
+
+    # for status codes see https://developers.google.com/maps/documentation/geocoding/intro#StatusCodes
+    raise GoogleMapsGeocoderErrorHandler::ZeroResultsError,\
+          message if status == 'ZERO_RESULTS'
+    raise GoogleMapsGeocoderErrorHandler::QueryLimitError,\
```

Comment:
I like your error-class names here, but do we really need the parent class `GoogleMapsGeocoderErrorHandler`? I'd rather extend `StandardError` directly as mentioned earlier.